MartinSmith-Lemures/hw2/Array_hw.rb

```diff
@@ -0,0 +1,58 @@
+    puts "Find the length of strings in an array"
+    
+# Version 1
+    def length_finder(input_array)
+      input_array.map! {|i| i.length}
+    end
+    
+#version 2
+    def length_finder(input_array)
+      destination = []
+      input_array.each do |i|
+        destination << i.length
+      end
+      p destination
+    end
+        
+    puts "Find the frequency of a string in a sentence"
+    
+#version 1
+    def find_frequency(sentence, word)
+      arr = sentence.upcase.split
+      word_count = arr.count(word.upcase)
+      #this should works I think just rubyMonk doesn't want fancy.
+      #puts "#{sentence}: has #{word_count} "#{word}"'s in it."
+    end
+    
+    puts "Select random elements from an array"
+    
+#version 1
+    def random_select(array, n)
+      array.shuffle.sample(n)
+    end
+    
+    
+    
+    Puts "Find non-duplicate values in an Array"
```

Comment:
Capital P in `puts` makes for flustered terminal output:
`Array_hw.rb:36:in '<main>': undefined method 'Puts' for main:Object (NoMethodError)`